src/qt/test/optiontests.cpp

```diff
@@ -122,3 +123,19 @@ void OptionTests::parametersInteraction()
     QVERIFY(!settings.contains("fListen"));
     gArgs.ClearPathCache();
 }
+
+void OptionTests::extractFilter()
+{
+    QString filter = QString("Partially Signed Transaction (Binary) (*.psbt)");
+    QCOMPARE(GUIUtil::extractFirstSuffixFromFilter(filter), "psbt");
+
+    filter = QString("Comma separated file (*.csv)");
+    QCOMPARE(GUIUtil::extractFirstSuffixFromFilter(filter), "csv");
+
+    filter = QString("PNG Image (*.png)");
+    QCOMPARE(GUIUtil::extractFirstSuffixFromFilter(filter), "png");
+
+    filter = QString("Wallet Data (*.dat)");
+    QCOMPARE(GUIUtil::extractFirstSuffixFromFilter(filter), "dat");
```

Comment:
Why 4 tests with the only suffix? Maybe just two tests? First test with one suffix, another one with two suffixes.